### ISSUE TITLE
Still alive?

### ISSUE THREAD

**Issue by angoras** [jackhumbert/FreeTheWheel#2]
Hi, this project is exactly what I was looking for, but is it still under development?

The ReadMe doesn't say anything about project status and compatibility..

**Comment by jackhumbert** [comment]
Still needs some work for full compatibility, but it's good enough for some uses right now.

I haven't had the chance to work on it much recently.

**Comment by abhibeckert** [comment]
@jackhumbert there's code commented out with a note about clutch support...

Why? I don't suppose the clutch will work if I just uncomment/recompile?

**Comment by jackhumbert** [comment]
I don't think so (been a while), but you're welcome to try!

**Comment by fdanapfel** [comment]
Hello,

if you (or somebody else) are still planning to extend this tool, it would be great if it could be turned into a more "Mac-like" application. It would be nice if it had a window where options can be configured and also the wheel functions can be tested (similar to https://github.com/360Controller/360Controller).

BTW. I've noticed that Logitech provides some documentation for the programming API for their wheels at https://opensource.logitech.com/opensource/index.php/Technical_Information (which seems to be down at the moment unfortunately, I have a copy of the document if somebody is interested), which can be useful for extending this tool. For example it seems to be possible to activate 900 degree rotation without having to switch the wheel to native mode, which makes it possible to use this option together with ForceFeedback.

**Comment by abhibeckert** [comment]
>  I have a copy of the document if somebody is interested

Yes please, can you attach a copy here?

**Comment by fdanapfel** [comment]
> Yes please, can you attach a copy here?

Here it is.
[Logitech_Force_Feedback_Protocol_V1.5.pdf](https://github.com/jackhumbert/FreeTheWheel/files/225374/Logitech_Force_Feedback_Protocol_V1.5.pdf)

**Comment by abhibeckert** [comment]
Thanks that looks really useful.

I haven't been on iRacing the last few months but when I get back into it I'll have a play with various different modes and see how I go.

**Comment by fdanapfel** [comment]
> Thanks that looks really useful.

Yes. To figure out how to change various settings it also helps to have a look at the source code of the LTWheelconf tool for Linux: https://github.com/TripleSpeeder/LTWheelConf
As you can see there it is actually possible to change a lot more settings than what is currently possible with FreeTheWheel.

**Comment by jfitzgerald** [comment]
Ran this on 10.9.5 to play Dirt 2 and it didn't seem to detect my Logitech Driving Force Pro (PS2).:

`Looking for supported wheels to enable in NATIVE mode. . .`

When I ran the --info option it shows this:

`Device ID=c294046d   Product ID=Logitech Driving Force (RESTRICTED)`

**Comment by jfitzgerald** [comment]
Apologies. I just realized this is not "the" Feral utility. My comment above is based on my use of their package.

**Comment by abhibeckert** [comment]
Andrew Eikum has ported the Logitech wheel driver for Linux to Mac, and it seems to work perfectly.

https://gitlab.com/ColdPie/logiwheel2000

Installation is complicated (you have to modify /System) but it works perfectly.

**Comment by fdanapfel** [comment]
@abhibeckert: sounds interesting! Since there is not much information on the gitlab page could you provide a link to the instructions on the iracing forums on how to set this up?

**Comment by abhibeckert** [comment]
iRacing forums require an active/paid account. It's easily found in the dedicated Mac OS C subforum section if you have access.

Essentially you need to disable system integrity protection and the requirement for kext's to be signed by Apple.

Then unload and remove Apple's Logitech wheel kext, put the new one in the same directory, and load it. Then it should "just work", although it has only been tested by a few people (and we all run the latest OS X and use the same wheel - G27.

When I get time I'll try to make an install script and full instructions.

It's a port from Linux that supports almost all Logitech wheels m, so it should work for you.

**Comment by ColdPie1** [comment]
> I'm not sure if I have permission to copy/paste them over here.

Go to town, my friend. In retrospect, I should put those installation instructions in the README.

**Comment by stevenvachon** [comment]
@ColdPie1 please do.

**Comment by ColdPie1** [comment]
@stevenvachon Your wish is my command ;) Further discussion on the issue you opened on the logiwheel2000 project.

**Comment by stevenvachon** [comment]
Nothing seems to work on my machine :confused:

**Comment by kevholt13** [comment]
Hi can any one help make a FreeTheWheel dmg file for mac that supports the G29 Wheel. Really need to get the g29 rotating 900 degrees like my old Gforce wheel did. Looks like someone has editted code. not sure how it get it to work. Thanks

**Comment by JJHackimoto** [comment]
> @jackhumbert there's code commented out with a note about clutch support...
> 
> Why? I don't suppose the clutch will work if I just uncomment/recompile?

Yes, uncommenting actually worked perfectly for me with my G27 on Catalina!

**Comment by digger3749** [comment]
Hi there.  Just got a second hand G27 and it wont work.  Tried running this and got the 900 degree steering however shifter and clutch not working.  Is this project still going?  I did a little programming a long time ago but would like to try and help with this one.

**Comment by JJHackimoto** [comment]
> Hi there. Just got a second hand G27 and it wont work. Tried running this and got the 900 degree steering however shifter and clutch not working. Is this project still going? I did a little programming a long time ago but would like to try and help with this one.

@digger3749 Did you try uncommenting the code as described above? It worked perfectly for me and I've been using this since then with my G27. Hopefully the developer here can see this and release it as a new version.

**Comment by digger3749** [comment]
> @digger3749 Did you try uncommenting the code as described above? It worked perfectly for me and I've been using this since then with my G27. Hopefully the developer here can see this and release it as a new version.

Been so long since I've coded.  Looking at it it's like looking at a different language.  So you've been able to get the whole kit to work.  Buttons, levers, shifter, etc with this code?

**Comment by JJHackimoto** [comment]
> > @digger3749 Did you try uncommenting the code as described above? It worked perfectly for me and I've been using this since then with my G27. Hopefully the developer here can see this and release it as a new version.
> 
> Been so long since I've coded. Looking at it it's like looking at a different language. So you've been able to get the whole kit to work. Buttons, levers, shifter, etc with this code?

Yes that's right! I tried to find the original files but seem to have deleted them after making the fix (and rebuilding in Xcode). The change was however made in the WheelSupports.cpp file. If my memory is correct, I uncommented the lines after "// Full button mapping with clutch" (line 372 to 379) and commented away the code above (362 to 369). I believe that was all I had to do. I still have the built version of the app but I'm not sure if I'm allowed to share it so please try the above and hope it works!

**Comment by digger3749** [comment]
> Yes that's right! I tried to find the original files but seem to have deleted them after making the fix (and rebuilding in Xcode). The change was however made in the WheelSupports.cpp file. If my memory is correct, I uncommented the lines after "// Full button mapping with clutch" (line 372 to 379) and commented away the code above (362 to 369). I believe that was all I had to do. I still have the built version of the app but I'm not sure if I'm allowed to share it so please try the above and hope it works!

Just tried it again and it didn't work.  All I need to do is run the executable file?  Does it need to be run with anything else like the old FreetheWheel program?

**Comment by JJHackimoto** [comment]
> > Yes that's right! I tried to find the original files but seem to have deleted them after making the fix (and rebuilding in Xcode). The change was however made in the WheelSupports.cpp file. If my memory is correct, I uncommented the lines after "// Full button mapping with clutch" (line 372 to 379) and commented away the code above (362 to 369). I believe that was all I had to do. I still have the built version of the app but I'm not sure if I'm allowed to share it so please try the above and hope it works!
> 
> Just tried it again and it didn't work. All I need to do is run the executable file? Does it need to be run with anything else like the old FreetheWheel program?

It will have to be recompiled using Xcode. Open the files in the Xcode app and all you have to do is basically to press build. This will create the app as an executable that you can then give an app icon etc. :)

**Comment by digger3749** [comment]
Thanks for the info.  Been trying to get my head around XCode and I really have no idea what I'm doing even to set up the project.  If I cerate a new Command Line project and then just drag and drop all the files in there, is that going to be enough?

**Comment by digger3749** [comment]
Tried again and get errors when building.  

Undefined symbols for architecture x86_64:
  "_CFArrayAppendValue", referenced from:
      SetApplierFunctionCopyToCFArray(void const*, void*) in WheelSupports.o
etc, etc, etc.

Pretty clear I have been out of the programming game for a long time.  If anyone can send me the updated files or help with running XCode I'd really appreciate it.

**Comment by JJHackimoto** [comment]
> Tried again and get errors when building.
> 
> Undefined symbols for architecture x86_64:
> "_CFArrayAppendValue", referenced from:
> SetApplierFunctionCopyToCFArray(void const*, void*) in WheelSupports.o
> etc, etc, etc.
> 
> Pretty clear I have been out of the programming game for a long time. If anyone can send me the updated files or help with running XCode I'd really appreciate it.

I can send the completed app (I can't remember how I solved the errors, but remember it involved a lot of googeling). I really hope I'm allowed to do this, and hopefully it will help others as well. Where can I post it?

**Comment by digger3749** [comment]
> I can send the completed app (I can't remember how I solved the errors, but remember it involved a lot of googeling). I really hope I'm allowed to do this, and hopefully it will help others as well. Where can I post it?

Thank you so much.  Any chance you can email me the zip file?  email address is digger3749@gmail.com
